lib/std/fifo.zig

```diff
@@ -376,6 +378,14 @@ pub fn LinearFifo(
     };
 }
 
+test "LinearFifo(u8, .Dynamic) discard(0) from empty buffer should not error on overflow" {
```

Comment:
Might be worth using a more succinct test name? Also the behaviour isn't unique to `u8` or `Dynamic`

Reply:
I'll open another PR soon to update / fix - I was out for the evening